src/DcmTrajectoryGenerator.cpp

```diff
@@ -0,0 +1,652 @@
+/**
+ * @file DcmTrajectoryGenerator.cpp
+ * @author Giulio Romualdi
+ * @copyright 2018 iCub Facility - Istituto Italiano di Tecnologia
+ *            Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ * @date 2018
+ */
+
+// std
+#include <math.h>
+#include <vector>
+#include <iostream>
+
+// eigen
+#include <Eigen/Dense>
+
+// iDynTree
+#include "iDynTree/Core/VectorFixSize.h"
+#include "iDynTree/Core/EigenHelpers.h"
+
+#include "DcmTrajectoryGenerator.h"
+
+GeneralSupportTrajectory::GeneralSupportTrajectory(const double &startTime, const double &endTime)
+{
+  // set the general support trajectory domain
+  m_trajectoryDomain = std::make_pair(startTime, endTime);
+}
+
+bool GeneralSupportTrajectory::timeBelongsToDomain(const double &t)
+{
+  // check if m_startTime <= t <=  m_endTime
+  double startTime = std::get<0>(m_trajectoryDomain);
+  double endTime = std::get<1>(m_trajectoryDomain);
+  
+  if ((t >= startTime) && (t <= endTime))
+    return true;
+
+  std::cerr << "[GENERAL SUPPORT TRAJECTORY] the time t: " << t << " does not belong to the trajectory domain" << std::endl; 
+  return false;
+}
+
+const std::pair<double, double> &GeneralSupportTrajectory::getTrajectoryDomain() const
+{
+  return m_trajectoryDomain;
+}
+
+SingleSupportTrajectory::SingleSupportTrajectory(const double &startTime,
+						 const double &endTime,
+						 const double &t0,
+						 const double &stepDuration,
+						 const double &omega,
+						 const iDynTree::Vector2 &zmp,
+						 const iDynTree::Vector2 &nextDcmIos):
+  GeneralSupportTrajectory(startTime, endTime),
+  m_t0(t0),
+  m_omega(omega),
+  m_zmp(zmp)
+{
+  setDcmIos(nextDcmIos, stepDuration);
+}
+
+void SingleSupportTrajectory::setDcmIos(const iDynTree::Vector2 &nextDcmIos,
+					const double &stepDuration)
+{
+  // The position of the DCM at the beginning of the i-th step
+  // depends on the position of the DCM at the beginning of the steps i+1
+  // dcm_ios[i] = dcm_ios[i+1]
+  //              + exp(-omega * Ts[i]) * (1 - exp(-omega * Ts[i])) * zmp[i]
+  
+  iDynTree::toEigen(m_dcmIos) = iDynTree::toEigen(m_zmp) + exp(-m_omega * stepDuration) * (iDynTree::toEigen(nextDcmIos) - iDynTree::toEigen(m_zmp));
```

Comment:
So you are assuming that the zmp is constant?

Reply:
Yes, during the _single support_ phase the position of the ZMP is assumed to be constant, while during the _double support_ phase the ZMP shifts from the center of the `SwichIn` foot to the center of the `SwitchOut` foot.

Reply:
Is this an assumption related to the fact that we are at planning level right?

Reply:
yes

Reply:
Ok got it!